src/governance/object.h

```diff
@@ -223,16 +224,16 @@ class CGovernanceObject
 
     // CORE OBJECT FUNCTIONS
 
-    bool IsValidLocally(std::string& strError, bool fCheckCollateral) const EXCLUSIVE_LOCKS_REQUIRED(cs_main);
+    bool IsValidLocally(const CDeterministicMNList& tip_mn_list, std::string& strError, bool fCheckCollateral) const EXCLUSIVE_LOCKS_REQUIRED(cs_main);
```

Comment:
consider create a new class such as `CGovernanceObjectVotable` which is:

```cpp
class CGovernanceObjectVotable
{
public:
    CGovernanceObject m_obj;
    const CDeterministicMNList tip_mn_list;
  CGovernanceObjectVotable(const CDeterministicMNList& mn_list, CGovernanceObject m_obj);
....
....

};
```

And move all methods that requires CDeterministicMNList such as IsValidLocally, UpdateLocalValidy, etc - to this new class.

In this case code below such as rpc implementation can be rewritten like that:
```
 const CGovernanceObjectVotable* pGovObj = Governance::MakeVotable(node.govman->FindConstGovernanceObject(hash), node.dmnman->GetListAtChainTip());

    UniValue objFundingResult(UniValue::VOBJ);
    objFundingResult.pushKV("AbsoluteYesCount",  pGovObj->GetAbsoluteYesCount(VOTE_SIGNAL_FUNDING));
    objFundingResult.pushKV("YesCount",  pGovObj->GetYesCount(VOTE_SIGNAL_FUNDING));
    objFundingResult.pushKV("NoCount",  pGovObj->GetNoCount(VOTE_SIGNAL_FUNDING));
    objFundingResult.pushKV("AbstainCount",  pGovObj->GetAbstainCount(VOTE_SIGNAL_FUNDING));
```

Reply:
I don't think this is needed in this PR; maybe a future PR